### PR TITLE
Get ready for automated beta release, enable all C.I. checks

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Same as: https://github.com/CrowdStrike/ember-toucan-core/pull/14

To exit pre-release mode:
```
pnpm changeset pre exit
```
Docs: https://github.com/changesets/changesets/blob/main/docs/prereleases.md 

Note that not every PR needs a changeset entry, and this is intentional -- especially as we work towards the baseline set of features.

But we do need at least one changeset to have our first beta release.
I've added the Changeset bot which provides instructions on how to make a changeset.

_We do not expect external contributors to provide changesets_.
The best workflow I've seen is for the PR reviewer (if contribution is external) to add the changeset to the PR before merge. 

As a fallback, I've made another tool, demonstrated over here: https://github.com/embroider-build/embroider/pull/1345 
which allows you to double check contributions before release.

---------------------------------

## What's included in this repo's C.I.?

 - assert that source maps and type-maps are generated
 - support multiple versions of typescript, per https://www.semver-ts.org/
 - support multiple versions of ember-source, to ensure a smooth upgrade path for downstream consumers
 - support embroider so that downstream consumers can be assured that their dependencies are "embroider strict mode safe"
- test with no lockfile to try to catch unintended ecosystem deviations / breaking changes (such as what may happen with babel / rollup / etc)

(all of the above is standard in ember OSS projects 🎉 )

extra stuff for this repo:
- publish the docs site and post the preview URLs to PRs
- configure the auto-release system to do a series of beta versions until we're ready for 1.0.0

## What is supported?

- minimum ember-source 4.8 -- we can start pretty new, but this also allows us to use the nicest features. Additionally, since we are beta, we can move the minimum ember-source to whichever ember-source ships the built-in-types (as non-preview), without a breaking change
- TS 4.8 and 4.9 -- we cannot support earlier than TS 4.8, because Glint does not support prior to TS 4.8
- embroider-strict and classic builds

-----------------------------------------

(some of this was already happening -- but this is copy-pasta from toucan-core)

### CODE REVIEW

File: ember-headless-form/package.json [hunk 1]
```diff
@@ -21,6 +21,9 @@
     "lint:hbs:fix": "ember-template-lint . --fix --no-error-on-unmatched-pattern",
     "lint:js:fix": "eslint . --fix",
     "lint:types": "glint",
+    "lint:prettier": "prettier -c .",
```
Comment (NullVoxPopuli) [inline ember-headless-form/package.json:24]: as an aside, I _really_ like the workflow of having so much namespaced under `lint`. makes the number of commands to remember a lot less, and the `pnpm --filter '*'` at the top makes seeing the output wonderful

File: ember-headless-form/tsconfig.json [hunk 2]
```diff
@@ -1,9 +1,6 @@
 {
   "extends": "@tsconfig/ember/tsconfig.json",
-  "include": [
```
Comment (NullVoxPopuli) [inline ember-headless-form/tsconfig.json:3]: prettier 🤷

File: test-app/package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
   "devDependencies": {
     "@ember/optional-features": "^2.0.0",
     "@ember/test-helpers": "^2.8.1",
-    "@embroider/test-setup": "^1.7.1",
+    "@embroider/test-setup": "^2.1.1",
```
Comment (NullVoxPopuli) [inline test-app/package.json:31]: this upgrade fixes the issue that embroiderOptimized/safe was handling before: https://github.com/embroider-build/embroider/blob/main/CHANGELOG.md#embroidertest-setup-202---210

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -11,52 +11,96 @@ concurrency:
...
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/pnpm
+      - uses: ./.github/actions/download-built-package
+      - run: pnpm --filter test-app test:ember
+
+  floating_tests:
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:86]: floating tests moved to here

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -11,52 +11,96 @@ concurrency:
...
+      # To be able to run glint, we need the dist directory
+      - uses: ./.github/actions/download-built-package
+      - name: Lint + Format + Glint
         run: pnpm lint
-      - name: Run Tests
-        run: pnpm test
 
-  floating:
-    name: 'Floating Dependencies'
+  typecheck:
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:48]: new to CI: typescript matrix support

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -11,52 +11,96 @@ concurrency:
...
+    name: Lint
+    runs-on: ubuntu-latest
+    needs: [install_dependencies, build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/pnpm
+      # To be able to run glint, we need the dist directory
+      - uses: ./.github/actions/download-built-package
+      - name: Lint + Format + Glint
         run: pnpm lint
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:46]: not all of our repos are doing an all-in-one lint command, but it's quite nice -- would be good to propagate this to everywhere

File: .changeset/README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# Changesets
+
+Hello and welcome! This folder has been automatically generated by `@changesets/cli`, a build tool that works
+with multi-package repos, or single-package repos to help you version and publish your code. You can
+find the full documentation for it [in our repository](https://github.com/changesets/changesets)
+
+We have a quick list of common questions to get you started engaging with this project in
+[our documentation](https://github.com/changesets/changesets/blob/main/docs/common-questions.md)
```
Comment (nicolechung) [inline .changeset/README.md:8]: Can we have a section here for beta releases?

```
## Beta releases


npx changeset-recover@beta


edit the generated files to tidy up the generate Changelog entry:
   - select whether the change is a bugfix (patch), minor change, or major / breaking change
   - edit the changelog entry, which defaults to a combination of both the PR title and PR description - and if an associated PR can't be found, the commit message will be used (useful for direct commits to main)
   ```

Comment (NullVoxPopuli) [reply .changeset/README.md:8]: changeset-recover is a fallback tool, for when the users of changeset don't want to use the changeset-recommended workflow. idk that it needs to be mentioned here.

All will be clear when we prepare for a release, I promise 🎉

**Review by nicolechung — APPROVED** [review]
🎉

File: .github/renovate.json5 [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    {
+      "groupName": "CLI Dependencies",
+      "packageNames": [
+        "broccoli-asset-rev",
+        "ember-cli",
+        "ember-auto-import",
+        "ember-cli-dependency-checker",
+        "ember-cli-inject-live-reload",
+        "ember-cli-sri",
+        "ember-cli-terser"
```
Comment (simonihmig) [inline .github/renovate.json5:73]: Same

File: .github/renovate.json5 [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    {
+      "groupName": "Testing Dependencies",
+      "schedule": ["after 9pm on sunday"],
+      "packageNames": [
+        "qunit-dom",
+        "ember-try",
+        "ember-source-channel-url",
+        "ember-qunit",
+        "qunit",
+        "npm-run-all"
```
Comment (simonihmig) [inline .github/renovate.json5:85]: Same

File: .github/renovate.json5 [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+      ]
+    },
+    {
+      // We will handle this ourselves
+      "groupName": "Automated Release Dependencies",
+      "enabled": false,
+      "packagePatterns": [
+        "@semantic-release*",
+        "semantic-release*"
```
Comment (simonihmig) [inline .github/renovate.json5:94]: Are we using those?

**Review by simonihmig — APPROVED** [review]
This is great, thanks for tackling this!

I just left a few nit-picking questions, as they came up within this context, but nothing that should block us from merging this!

File: .github/renovate.json5 [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+        "eslint-config-.*",
+        ".*typescript-eslint.*",
+        "^@commitlint\/*",
+        "^remark-*"
+      ]
+    },
+    // These are dependencies that come default when
+    // generating a new ember addon
+    {
+      "groupName": "Framework Dependencies",
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply .github/renovate.json5:50]: Just less PRs if they do happened to get released close to each other, or if we're falling behind on updates. They tend to not cause problems for one another as far as I've been able to tell.

File: .github/renovate.json5 (hunk 10 above)
Comment (NullVoxPopuli) [reply .github/renovate.json5:94]: nope! this is copy-pasta

File: .github/renovate.json5 [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+    },
+    {
+      // Max Semver compatibility
+      "packagePatterns": [
+        "@ember/test-waiters"
+      ],
+      "rangeStrategy": "widen"
+    },
+    {
+      // changing peerDependencies *at all* is a breaking change
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply .github/renovate.json5:105]: auto-merging dev stuff is fine, but auto-merging peers affects consumers.

By default, renovate updates everything, with no respect to semver.

> major bump raise a PR

this is a very nice layout to the renovate config.
I'd be open to creating a similar shared set of configs for our ember stuff, if you wanted to set that up?
And `automerge: false` is totally something we can do on certain things!

File: .github/renovate.json5 [hunk 13]
```diff
@@ -0,0 +1,122 @@
+// Docs:
+// https://docs.renovatebot.com/configuration-options/
+{
+  "extends": [
+    "config:base"
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply .github/renovate.json5:5]: > but are we always adding the whole config for each repo?

yeah, we don't have a hosted config yet. sounds like a great idea!

File: .github/workflows/ci.yml [hunk 14]
```diff
@@ -67,22 +111,114 @@ jobs:
...
 
     steps:
       - uses: actions/checkout@v3
-      - name: Install pnpm
-        uses: pnpm/action-setup@v2
         with:
-          version: 7
-      - name: Install Node
-        uses: actions/setup-node@v3
+          persist-credentials: false
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply .github/workflows/ci.yml:143]: not persisting-credentials sounds nice. This is copy-paste from the Changesets/action README.md. 
I don't have any reasons other than trust

**Pull request merged**